### PR TITLE
[Enhancement] Support remove immutable memtable for cloud native pindex

### PR DESCRIPTION
**Pull request by sevev**
## Why I'm doing:
Cloud native persistent index memtable support multiple memtables include one mutable memtable and multiple immutable memtables. However, multiple memtables will use more memory and generate many small sst files which increase read cost. Apart from that, flushing the memtable is not the bottleneck of the apply process, so having a single memtable is more appropriate.

## What I'm doing:
Support single memtable for cloud native persistent index.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -1522,4 +1522,6 @@ CONF_mBool(avro_ignore_union_type_tag, "false");
 // default batch size for simdjson lib
 CONF_mInt32(json_parse_many_batch_size, "1000000");
 CONF_mBool(enable_dynamic_batch_size_for_json_parse_many, "true");
+
+CONF_Bool(enable_multiple_memtable_for_lake_pindex, "false");
```
Comment (luohaha) [inline be/src/common/config.h:1526]: I don't think we need to make it configurable.

Comment (luohaha) [reply be/src/common/config.h:1526]: And remove `_immutable_memtable`

File: be/src/storage/lake/lake_persistent_index.cpp [hunk 2]
```diff
@@ -243,23 +242,12 @@ Status LakePersistentIndex::get_from_sstables(size_t n, const Slice* keys, Index
...
-}
-
 Status LakePersistentIndex::get(size_t n, const Slice* keys, IndexValue* values) {
     KeyIndexSet not_founds;
     // Assuming we always want the latest value now
     RETURN_IF_ERROR(_memtable->get(n, keys, values, &not_founds, -1));
     KeyIndexSet& key_indexes = not_founds;
     KeyIndexSet found_key_indexes;
-    RETURN_IF_ERROR(get_from_immutable_memtable(keys, values, key_indexes, &found_key_indexes, -1));
     set_difference(&key_indexes, found_key_indexes);
```
Comment (wyb) [inline be/src/storage/lake/lake_persistent_index.cpp:251]: `set_difference` not used

Comment (sevev) [reply be/src/storage/lake/lake_persistent_index.cpp]: > `set_difference` not used

done

File: be/src/common/config.h (hunk 1 above)
Comment (sevev) [reply be/src/common/config.h]: > I don't think we need to make it configurable.

done

**Review by wyb — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Pull request merged**